### PR TITLE
APL records: fix endianness problem. Closes #9763

### LINKED ISSUE #9763: APL tests fail on Big Endian machines
- Program: Recursor 
 - Issue type: Bug report

### Short description

```
check rec2->getZoneRepresentation() == zoneval has failed
[1:0.0.0.10/32 2:100::/8 != 1:10.0.0.0/32 2:100::/8] 
```

### Environment

 - Operating system: Debian
 - Software version: 4.4.0
 - Software source: self compiled (during package build)

### Steps to reproduce

1. build on big endian machines:
 s390x logs: https://buildd.debian.org/status/fetch.php?pkg=pdns-recursor&arch=s390x&ver=4.4.0-1&stamp=1606184595&file=log
 ppc64 logs: https://buildd.debian.org/status/fetch.php?pkg=pdns-recursor&arch=ppc64&ver=4.4.0-1&stamp=1606172842&raw=0

### Expected behaviour

Tests complete.

### Actual behaviour

Tests fail:

```
Checking record type APL test #1 
test-dnsrecords_cc.cc(251): error: in "test_dnsrecords_cc/test_record_types": check rec->getZoneRepresentation() == zoneval has failed [1:0.0.0.10/32 != 1:10.0.0.0/32] 
test-dnsrecords_cc.cc(259): error: in "test_dnsrecords_cc/test_record_types": check rec2->getZoneRepresentation() == zoneval has failed [1:0.0.0.10/32 != 1:10.0.0.0/32] 
Checking record type APL test #2 
test-dnsrecords_cc.cc(251): error: in "test_dnsrecords_cc/test_record_types": check rec->getZoneRepresentation() == zoneval has failed [1:1.1.1.10/32 != 1:10.1.1.1/32] 
test-dnsrecords_cc.cc(259): error: in "test_dnsrecords_cc/test_record_types": check rec2->getZoneRepresentation() == zoneval has failed [1:1.1.1.10/32 != 1:10.1.1.1/32] 
Checking record type APL test #3 
test-dnsrecords_cc.cc(251): error: in "test_dnsrecords_cc/test_record_types": check rec->getZoneRepresentation() == zoneval has failed [1:0.1.1.0/24 != 1:10.1.1.0/24] 
test-dnsrecords_cc.cc(259): error: in "test_dnsrecords_cc/test_record_types": check rec2->getZoneRepresentation() == zoneval has failed [1:0.1.1.0/24 != 1:10.1.1.0/24] 
Checking record type APL test #4 
test-dnsrecords_cc.cc(251): error: in "test_dnsrecords_cc/test_record_types": check rec->getZoneRepresentation() == zoneval has failed [1:0.0.0.0/8 != 1:60.0.0.0/8] 
test-dnsrecords_cc.cc(259): error: in "test_dnsrecords_cc/test_record_types": check rec2->getZoneRepresentation() == zoneval has failed [1:0.0.0.0/8 != 1:60.0.0.0/8] 
Checking record type APL test #5 
test-dnsrecords_cc.cc(251): error: in "test_dnsrecords_cc/test_record_types": check rec->getZoneRepresentation() == zoneval has failed [1:1.1.1.10/32 != 1:10.1.1.1/32] 
test-dnsrecords_cc.cc(259): error: in "test_dnsrecords_cc/test_record_types": check rec2->getZoneRepresentation() == zoneval has failed [1:1.1.1.10/32 != 1:10.1.1.1/32] 
Checking record type APL test #6 
test-dnsrecords_cc.cc(251): error: in "test_dnsrecords_cc/test_record_types": check rec->getZoneRepresentation() == zoneval has failed [!1:1.1.1.10/32 != !1:10.1.1.1/32] 
test-dnsrecords_cc.cc(259): error: in "test_dnsrecords_cc/test_record_types": check rec2->getZoneRepresentation() == zoneval has failed [!1:1.1.1.10/32 != !1:10.1.1.1/32] 
Checking record type APL test #7 
Checking record type APL test #8 
Checking record type APL test #9 
Checking record type APL test #10 
Checking record type APL test #11 
Checking record type APL test #12 
Checking record type APL test #13 
Checking record type APL test #14 
Checking record type APL test #15 
Checking record type APL test #16 
Checking record type APL test #17 
Checking record type APL test #18 
test-dnsrecords_cc.cc(251): error: in "test_dnsrecords_cc/test_record_types": check rec->getZoneRepresentation() == zoneval has failed [1:0.0.0.10/32 1:1.1.1.10/32 != 1:10.0.0.0/32 1:10.1.1.1/32] 
test-dnsrecords_cc.cc(259): error: in "test_dnsrecords_cc/test_record_types": check rec2->getZoneRepresentation() == zoneval has failed [1:0.0.0.10/32 1:1.1.1.10/32 != 1:10.0.0.0/32 1:10.1.1.1/32] 
Checking record type APL test #19 
test-dnsrecords_cc.cc(251): error: in "test_dnsrecords_cc/test_record_types": check rec->getZoneRepresentation() == zoneval has failed [1:0.0.0.10/32 2:100::/8 != 1:10.0.0.0/32 2:100::/8] 
test-dnsrecords_cc.cc(259): error: in "test_dnsrecords_cc/test_record_types": check rec2->getZoneRepresentation() == zoneval has failed [1:0.0.0.10/32 2:100::/8 != 1:10.0.0.0/32 2:100::/8] 
```

### PR DESCRIPTION
**Pull request by Habbie**
### Short description
Tested on amd64 and s390x

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)
- [ ]  checked that this code was merged to master

### CODE REVIEW

**Review by zeha — APPROVED** [review]
lgtm

File: pdns/dnsrecords.cc [hunk 1]
```diff
@@ -694,20 +694,14 @@ string APLRecordContent::getZoneRepresentation(bool noDot) const {
...
       ca = ComboAddress();
       ca.sin4.sin_family = AF_INET6;
-      for (int i=0; i < 16; i++) {
-        if (i < ard->d_afdlength) {
-          ca.sin6.sin6_addr.s6_addr[i] = ard->d_ip.d_ip6[i];
-        } else {
-          ca.sin6.sin6_addr.s6_addr[i] = 0;
-        }
-      }
+      memset(&ca.sin6.sin6_addr.s6_addr, 0, 16);
```
Comment (rgacogne) [inline pdns/dnsrecords.cc:703]: Just a very small nit but perhaps:
```suggestion
      memset(&ca.sin6.sin6_addr.s6_addr, 0, sizeof(ca.sin6.sin6_addr.s6_addr));
```

**Review by rgacogne — APPROVED** [review]

File: pdns/dnsrecords.cc (hunk 1 above)
Comment (omoerbeek) [reply pdns/dnsrecords.cc:703]: But if you take the suggestion also do it for the ipv4 case...

Comment (Habbie) [reply pdns/dnsrecords.cc:703]: makes sense, in a few more places perhaps even!

**Comment by Habbie** [comment]
@threatstop hi! Just in case you're using this same code in other places, please be aware of this fix :)

**Comment by nicko2n** [comment]
> @threatstop hi! Just in case you're using this same code in other places, please be aware of this fix :)

Thanks Habbie

**Pull request merged**